### ISSUE TITLE
Strict parsing not as strict as it should be

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google/gson#499]
```
According to the docs, in strict parsing "Names that are unquoted" should throw 
an error, however the following code parses the json perfectly fine, although 
lenient is false:
new JsonParser().parse("{invalid: 1}");

I am using gson-2.1.
```

Original issue reported on code.google.com by `rum...@google.com` on 29 Jan 2013 at 10:45

**Comment by vkopichenko** [comment]
Here is a workaround for Gson strict parsing with many details and extensive test case: https://stackoverflow.com/questions/43233898/how-to-check-if-json-is-valid-in-java-using-gson/47890960#47890960

**Comment by Marcono1234** [comment]
This has been solved in Gson version 2.11.0 with the new `Strictness` API, but requires explicit opt-in, see https://github.com/google/gson/issues/372#issuecomment-1666258082.
